fish_prompt.fish

```diff
@@ -801,6 +802,20 @@ function __bobthefish_prompt_desk -S -d 'Display current desk environment'
     set_color normal
 end
 
+function __bobthefish_prompt_nix -S -d 'Display current nix environment'
+    [ "$theme_display_nix" = 'no' -o -z "$IN_NIX_SHELL" ]
+    and return
+
+    if [ "$IN_NIX_SHELL" = "impure" ]
+        set say_what "nix"
+    else
+        set say_what "pure"
```

Comment:
bobthefish biases towards removing info when things are in the default state, and only showing them when they're "interesting". i don't know enough about nix to know whether this is a case like that, but it seems like it might be. should we be leaving off either the "pure" or "impure" side?